wordpress/mod.ts

```diff
@@ -36,174 +127,137 @@ export default function (userOptions?: Partial<Options>) {
 }
 
 export class WordPressAPI {
-  #main_endpoint: URL;
-  #api_endpoint: URL;
-  #limit: number;
-  // deno-lint-ignore no-explicit-any
-  #cache = new Map<string, any>();
+  static #apiRoute = "rest_route";
 
-  constructor(options: Options) {
-    this.#main_endpoint = new URL(options.wp_url);
-    this.#api_endpoint = new URL("./wp-json/wp/v2/", options.wp_url);
-    this.#limit = options.limit;
-  }
+  #base: URL;
+  #pass: string | undefined;
 
-  async info(): Promise<WP_Info> {
-    const res = await fetch(new URL("meta", this.#main_endpoint));
-    return await res.json() as WP_Info;
-  }
+  #limit: number;
+  #maxPerPage: number;
 
-  /** Returns all posts of the site */
-  async *posts(limit?: number): AsyncGenerator<Partial<PageData>> {
-    for await (const post of this.#fetchAll<WP_REST_API_Post>("posts", limit)) {
-      yield postToData(post);
-    }
-  }
+  #collections: Options["collections"];
+  #customTransformer: Transform;
 
-  /** Returns all pages of the site */
-  async *pages(limit?: number): AsyncGenerator<Partial<PageData>> {
-    for await (const page of this.#fetchAll<WP_REST_API_Post>("pages", limit)) {
-      yield pageToData(page);
-    }
-  }
+  #cache = new Map<string, unknown>();
 
-  /** Returns all categories of the site */
-  async *categories(limit?: number): AsyncGenerator<Partial<PageData>> {
-    for await (
-      const category of this.#fetchAll<WP_REST_API_Term>("categories", limit)
-    ) {
-      yield termToData(category);
-    }
-  }
+  constructor(options: Options) {
+    this.#base = new URL(options.baseUrl);
+    this.#limit = options.limit;
+    this.#maxPerPage = options.maxPerPage;
+    this.#collections = options.collections;
+    this.#customTransformer = options.transform;
 
-  /** Returns all tags of the site */
-  async *tags(limit?: number): AsyncGenerator<Partial<PageData>> {
-    for await (const tag of this.#fetchAll<WP_REST_API_Term>("tags", limit)) {
-      yield termToData(tag);
+    if (options.auth) {
+      this.#pass = options.auth.includes(":")
+        ? btoa(options.auth)
+        : options.auth;
     }
   }
 
-  /** Returns all authors of the site */
-  async *authors(limit?: number): AsyncGenerator<Partial<PageData>> {
-    for await (const user of this.#fetchAll<WP_REST_API_User>("users", limit)) {
-      yield userToData(user);
-    }
-  }
+  async *collection(
+    name: string,
+    limit: number = this.#limit,
+  ): AsyncGenerator<Partial<PageData>> {
+    const filterList: Transformer[] = [
+      requirementFilter,
+      presetTransform[name] || this.#customTransformer["*"],
+      this.#customTransformer[name],
```

Comment:
I think `this.#customTransformer[name]` could be also array, so people could add more transformers on top of the previous. Maybe something like:

```js
const customTransformer = Array.isArray(this.#customTransformer[name])
  ? this.#customTransformer[name]
  : [this.#customTransformer[name]];
```

Reply:
I thought about it too but I haven't came up a case where dev will stack transformers like that. If they need to do that, why don't they just put in 1 transformer cb? They have to type it down anyway.

Reply:
I'm thinking of allowing to create other plugins to modify the wp plugin. For example:

```js
site.use(wp()); // The wp standard plugin
site.use(woocomerce()); // Modify existing transformers or register new transformers addapted for WooCommerce sites.
site.use(wpMedia()); // To download the images
```

These plugins other plugins could add new elements (transformers) in the array, enhancing the default behavior.

Reply:
Solid reason, I'm all in.

Reply:
Okay, but internally it should be stored as an array in order to push more elements later.